src/vm/analysis/type_checker/natives/iterables.rs

```diff
@@ -0,0 +1,200 @@
+    check_argument_count(2, args)?;
+    
+    let lhs_type = checker.type_check(&args[0], context)?;
+    match lhs_type {
+        TypeSignature::ListType(lhs_list) => {
+            let rhs_type = checker.type_check(&args[1], context)?;
+            if let TypeSignature::ListType(rhs_list) = rhs_type {
+                if lhs_list.entry_type.admits_type(&*rhs_list.entry_type) {
+                    let return_type = TypeSignature::list_of(*lhs_list.entry_type, lhs_list.max_len + rhs_list.max_len)?;
+                    return Ok(return_type);
+                } else {
+                    return Err(CheckErrors::TypeError(*lhs_list.entry_type, *rhs_list.entry_type).into());
+                }
+            } else {
+                return Err(CheckErrors::TypeError(rhs_type.clone(), TypeSignature::ListType(lhs_list)).into());
+            }
+        },
+        TypeSignature::BufferType(lhs_buff_len) => {
+            let rhs_type = checker.type_check(&args[1], context)?;
+            if let TypeSignature::BufferType(rhs_buff_len) = rhs_type {
+                let size: u32 = u32::from(lhs_buff_len) + u32::from(rhs_buff_len);
+                let return_type = TypeSignature::buffer_of_size(size);
+                return Ok(return_type);
+            } else {
+                return Err(CheckErrors::TypeError(rhs_type.clone(), TypeSignature::max_buffer()).into());
+            }
+        },
+        _ => Err(CheckErrors::ExpectedListOrBuffer(lhs_type.clone()).into())
+    }
+}
+
+pub fn check_special_append(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(2, args)?;
+
+    let lhs_type = checker.type_check(&args[0], context)?;
+    match lhs_type {
+        TypeSignature::ListType(lhs_list) => {
+            let rhs_type = checker.type_check(&args[1], context)?;
+            if lhs_list.entry_type.admits_type(&rhs_type) {
+                let return_type = TypeSignature::list_of(*lhs_list.entry_type, lhs_list.max_len + 1)?;
+                return Ok(return_type);
+            } else {
+                return Err(CheckErrors::TypeError(*lhs_list.entry_type, rhs_type).into());
+            }
+        },
+        _ => Err(CheckErrors::ExpectedListApplication.into())
+    }
+}
+
+pub fn check_special_asserts_max_len(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(2, args)?;
+
+    let expected_len = match args[1].expr {
+        SymbolicExpressionType::LiteralValue(Value::UInt(expected_len)) => expected_len,
+        _ => {
+            let expexted_len_type = checker.type_check(&args[0], context)?;
+            return Err(CheckErrors::TypeError(TypeSignature::UIntType, expexted_len_type).into())
+        }
+    };
+    if expected_len > u128::from(MAX_VALUE_SIZE)  {
+        return Err(CheckErrors::MaxLengthOverflow.into())
+    }
+    let expected_len = expected_len as u32;
```

Comment:
You can replace those 4 lines with

```
let expected_len = u32::try_from(expected_len)
   .ok_or_else(CheckErrors::MaxLengthOverflow)?;
```

It's a little more concise, but also avoids creating a new u128 value for just doing a length check.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/commit/5eb2a3e13c04c5ac846095bc0cd59c973c94fa48